### PR TITLE
Expose DebugOptions and LogLevel through torch_ort

### PR DESCRIPTION
**Pull request by baijumeswani**
This pull request:
- Updates ```__init__.py``` to expose ```DebugOptions``` and ```LogLevel``` through ```torch_ort```.
- Updates ```experimental``` ```__init__.py``` to expose ```load_from_json``` function.
- Adds unit tests for ```DebugOptions``` and ```experimental.json_config.load_from_json```.

### CODE REVIEW

File: torch_ort/experimental/__init__.py [hunk 1]
```diff
@@ -7,9 +7,3 @@
 
 DO NOT use these APIs for any production code as they can be removed at any time
 """
-
-from .debug import save_intermediate_onnx_graphs, set_log_level
-from .graph_config import set_propagate_cast_ops_optimization,\
-                          PropagateCastOpsStrategy,\
-                          PropagateCastLevel
```
Comment (thiagocrepaldi) [inline torch_ort/experimental/__init__.py:14]: Why were these deleted? The `debug` part seems ok, but the propagate cast config will only be configurable through json, which is not ideal.

File: tests/experimental/torch_ort_experimental_basic_tests.py [hunk 2]
```diff
@@ -1,203 +0,0 @@
-import os
```
[the remark this answers is not in the document]
Comment (baijumeswani) [reply tests/experimental/torch_ort_experimental_basic_tests.py:1]: these tests are testing ```save_intermediate_onnx_graphs```, ```set_log_level``` and ```set_propagate_cast_ops_optimization``` functions. The first two are available through the ```DebugOptions``` in ```ORTModule``` and the last one is available through the ```load_from_json``` method in ```ORTModule``` experimental. So, I thought we could probably add tests for those methods instead. Moreover, these methods need slight modification to work with the updated ```ORTModule``` (due to attributes moving around).

If we think we should have both sets of tools to update the debug flags, I will keep the tests and update the functions.

File: tests/torch-ort_test_api.py [hunk 3]
```diff
@@ -34,11 +35,103 @@ def test_set_seed():
...
+    assert os.path.exists(os.path.join(os.getcwd(), f"my_cwd_model_optimized_{mode}.onnx"))
+
+    os.remove(os.path.join(os.getcwd(), f"my_cwd_model_torch_exported_{mode}.onnx"))
+    os.remove(os.path.join(os.getcwd(), f"my_cwd_model_optimized_{mode}.onnx"))
+
+def test_debug_options_save_onnx_models_validate_fail_on_non_writable_dir():
+
+    os.environ['ORTMODULE_SAVE_ONNX_PATH'] = '/non/existent/directory'
+    with pytest.raises(Exception) as ex_info:
+        _ = DebugOptions(save_onnx=True, onnx_prefix='my_model')
```
[the remark this answers is not in the document]
Comment (baijumeswani) [reply tests/torch-ort_test_api.py:90]: I'll update to use ```os.path.join```. I am not sure how we can assume that the ```tmp_dir``` does not exist in the example you shared.

File: torch_ort/experimental/__init__.py (hunk 1 above)
Comment (baijumeswani) [reply torch_ort/experimental/__init__.py:14]: I'll re-add that and update the function so it points to the correct attribute when changing the option.

File: torch_ort/experimental/debug.py [hunk 4]
```diff
@@ -1,75 +0,0 @@
-# -------------------------------------------------------------------------
```
[the remark this answers is not in the document]
Comment (baijumeswani) [reply torch_ort/experimental/debug.py:1]: Deleted because of introduction of ```DebugOptions```. I'll re-add it and update the functions if needed.

File: torch_ort/experimental/graph_config.py [hunk 5]
```diff
@@ -1,68 +0,0 @@
-# -------------------------------------------------------------------------
```
[the remark this answers is not in the document]
Comment (baijumeswani) [reply torch_ort/experimental/graph_config.py:1]: Deleted because of the ```load_from_json``` option to update propagate cast ops. I'll re-add this and update the functions if needed.

File: tests/torch-ort_test_api.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (thiagocrepaldi) [reply tests/torch-ort_test_api.py:90]: https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory creates a folder that is guaranteed to not conflict with any existing one. After `with` is closed, that folder is deleted, so the path that is saved is guaranteed to be non existent

File: torch_ort/experimental/debug.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (baijumeswani) [reply torch_ort/experimental/debug.py:1]: Removing code related to saving onnx files and setting log level as they are available through ```DebugOptions```.

File: torch_ort/experimental/__init__.py (hunk 1 above)
Comment (baijumeswani) [reply torch_ort/experimental/__init__.py:14]: I left this deleted because I thought it might make sense to keep the behavior consistent with ```json_config```. We want users to use these APIs as such:
```py
from torch_ort.experimental.graph_config import set_propagate_cast_ops_optimization, PropagateCastOpsStrategy, PropagateCastLevel
from torch_ort.experimental.json_config import load_from_json
```

Please let me know if you think this should change.

File: tests/experimental/torch_ort_experimental_basic_tests.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (baijumeswani) [reply tests/experimental/torch_ort_experimental_basic_tests.py:1]: deleted tests related to save onnx and setting log level. Moved propagate cast ops related tests to its own file: ```tests/experimental/torch_ort_graph_config_tests.py```

File: torch_ort/experimental/graph_config.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (baijumeswani) [reply torch_ort/experimental/graph_config.py]: undeleted it and updated the methods.

**Review by SherlockNoMad — APPROVED** [review]

**Review by liqunfu — COMMENTED** [review]
where are the new tests called? shall it be invoked by build.py?

**Review by liqunfu — APPROVED** [review]
talked with author offline. the tests are executed in other CI pipelines.

**Comment by baijumeswani** [comment]
we don't need to run these tests in the packaging pipeline. If we do set up the pipeline for CI, then we should probably add these tests to that.
